### PR TITLE
Improvements to package dependency lists

### PR DESCRIPTION
**Pull request by kislyuk**
Improvements to Python package dependencies

   - Add comments noting best practices for dependency management
   - Format dependency version ranges
   - Remove transitive (sub) dependencies
   - Update dependency minimum versions to those distributed by Ubuntu 18.04 where available
   - Bound version ranges in a consistent manner and leave comments where pins are necessary

Also:
   -  Fix lint issues and un-silence linter on many errors

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -2,4 +2,5 @@
 universal=1
 [flake8]
 max-line-length=120
-ignore: E301, E302, E401, E261, E265, E226, F401, E501
+ignore: F401, W504
+#ignore: E301, E302, E401, E261, E265, E226, F401, E501, W504
```
Comment (jessebrennan) [inline setup.cfg]: Looks like this stray comment snuck back in.

**Comment by jessebrennan** [comment]
@kislyuk woops... I didn't see this was merged when I started reviewing. Anyway, my comment was minor.

File: requirements.txt [hunk 2]
```diff
@@ -1,22 +1,37 @@
...
+google-auth-oauthlib >= 0.4.1, < 2
+Jinja2 >= 2.10, < 3
 jsonpointer >= 1.10, < 2
 jsonschema >= 2.6, < 3
-puremagic < 1.5
-PyJWT >= 1.6.4
+
+# Pinned due to an incompatibility with later versions of puremagic.
+# See https://github.com/HumanCellAtlas/dcp-cli/pull/245.
+puremagic==1.5
```
Comment (charlesreid1) [inline requirements.txt]: This version bump is now in conflict with dcplib: https://github.com/HumanCellAtlas/dcplib/blob/master/requirements.txt#L4

Comment (kislyuk) [reply requirements.txt]: @charlesreid1  that was an oversight on my part, thanks for calling it out.

Fixed in https://github.com/HumanCellAtlas/dcp-cli/commit/b91112a9389d9addb96417fd01481db562c1a3cf.

File: setup.cfg (hunk 1 above)
Comment (kislyuk) [reply setup.cfg]: Thanks, fixed it before merging.

**Pull request merged**